### ISSUE TITLE
Can't get intellisense to link params for anonymous functions

### ISSUE THREAD

**Issue by egamma** [Microsoft/TypeScript#9755]
_From @Sequoia on July 14, 2016 14:7_

- VSCode Version: Version 1.3.1 (1.3.1)
- OS Version: osx 10.11.5 (15F34)

## Goal

```js
return sftp.readdirAsync(dir)
  .then(function(/*Array<String>*/dirList) {
    dirList // <-- I want intellisense hinting on this array
```

## Expected
<img width="314" alt="screenshot 2016-07-14 10 02 04" src="https://cloud.githubusercontent.com/assets/317498/16842074/0e2baf18-49aa-11e6-861d-e4ba747923d6.png">

## Actual
<img width="287" alt="screenshot 2016-07-14 10 02 49" src="https://cloud.githubusercontent.com/assets/317498/16842103/26229730-49aa-11e6-8a2c-d3223789ea1b.png">

## Things I've tried

1. `.then(function(/*Array<String>*/dirList ) {`
1. `.then(function(/**Array<String>*/dirList ) {`
1. `.then(function(/**@type Array<String>*/dirList ) {`
1. `.then(function(/**@type {Array<String>}*/dirList ) {`
1. `.then(function(/**@type {String[]}*/dirList ) {`
1. `.then(/**@param {String[]} dirList */ function(dirList ) {`
1. `.then(function(dirList/**=String[]*/) {`
1. `.then(function(dirList/*=String[]*/) {`
1. `.then(function(dirList/*:String[]*/) {`
1. `.then(function(/**{String[]}:*/dirList ) {`
1. `.then(function(/*String[]:*/dirList ) {`

😢 Help

_Copied from original issue: Microsoft/vscode#9288_

**Comment by egamma** [comment]
_From @Sequoia on July 14, 2016 14:11_

readdirAsync is result of promisifyAll so not easily doc-linkable. (i.e. I can't easily mark the `@return {Promise<String[]>}` on that method).

**Comment by egamma** [comment]
@Sequoia in general the supported jsdoc annotation are defined in this [doc](https://github.com/Microsoft/TypeScript/wiki/JsDoc-support-in-JavaScript).

This form works for me:
```js
 sftp.readdirAsync(dir)
  .then(
      /** @param {string} dirList*/function (dirList) {
```

This form does **not** work for me (open paren in front of jsdoc comment):
```js
 sftp.readdirAsync(dir)
  .then
      (/** @param {string} dirList*/function (dirList) {
```

Moving to TypeScript for comment.

**Comment by mhegazy** [comment]
The second form that @egamma mentioned not working is due to https://github.com/Microsoft/TypeScript/issues/7533. this has since been fixed.

so:

```ts
return sftp.readdirAsync(dir)
.then(/**@param {String[]} dirList */ function(dirList ) {
    dirList.
```

should work.

**Comment by mhegazy** [comment]
I should note that the fix for this is part of TypeSript 2.1. VSCode ships with an older version of TypeScript. Please see [Using Newer TypeScript Versions](https://code.visualstudio.com/docs/languages/typescript#_using-newer-typescript-versions) documentation for more details on updating your VSCode to use a different version of TypeScript.
